File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/Car.java
package com.raxim.myscoutee.profile.data.document.mongo;

import java.util.Date;
import java.util.List;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Document(collection = "cars")
public class Car implements Cloneable {
    @Id
    @JsonProperty(value = "key")
    private UUID id;

    @JsonProperty(value = "regNum")
    private String regNum;

    @JsonProperty(value = "make")
    private String make;

    @JsonProperty(value = "model")
    private String model;

    @JsonProperty(value = "color")
    private String color;

    @JsonProperty(value = "year")
    private Integer year;

    @JsonProperty(value = "capacity")
    private Integer capacity;

    @JsonProperty(value = "images")
    private List<Image> images;

    @JsonIgnore
    private Date createdDate;

    @JsonIgnore
    private String status;

    public Car() {
    }

    public Car(UUID id) {
        this.id = id;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getRegNum() {
        return regNum;
    }

    public void setRegNum(String regNum) {
        this.regNum = regNum;
    }

    public String getMake() {
        return make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }

    public Integer getCapacity() {
        return capacity;
    }

    public void setCapacity(Integer capacity) {
        this.capacity = capacity;
    }

    public List<Image> getImages() {
        return images;
    }

    public void setImages(List<Image> images) {
        this.images = images;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public Object clone()
            throws CloneNotSupportedException {
        return super.clone();
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/Car.kt
package com.raxim.myscoutee.profile.data.document.mongo

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.querydsl.core.annotations.QueryEntity
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@QueryEntity
@Document(collection = "cars")
data class Car(
    @Id
    @field:JsonProperty(value = "key")
    val id: UUID? = UUID.randomUUID(),
    @field:JsonProperty(value = "regNum")
    val regNum: String? = null,
    @field:JsonProperty(value = "make")
    val make: String? = null,
    @field:JsonProperty(value = "model")
    val model: String? = null,
    @field:JsonProperty(value = "color")
    val color: String? = null,
    @field:JsonProperty(value = "year")
    val year: Int? = null,
    @field:JsonProperty(value = "capacity")
    val capacity: Int? = null,
    @field:JsonProperty(value = "images")
    val images: List<Image>? = emptyList(),

    @field:JsonIgnore
    var createdDate: Date? = Date(),

    //Active (A), Deleted (D)
    @field:JsonIgnore
    var status: String? = "A"
)